examples/ogbn_train.py

```diff
@@ -0,0 +1,266 @@
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter, )
+parser.add_argument(
+    '--dataset',
+    type=str,
+    default='ogbn-papers100M',
+    choices=['ogbn-papers100M', 'ogbn-products'],
+    help='Dataset name.',
+)
+parser.add_argument(
+    '--dataset_dir',
+    type=str,
+    default='./data',
+    help='Root directory of dataset.',
+)
+parser.add_argument(
+    "--dataset_subdir",
+    type=str,
+    default="ogb-papers100M",
+    help="directory of dataset.",
+)
+parser.add_argument(
+    '--use_gat',
+    action='store_true',
+    help='Whether or not to use GAT model',
+)
+parser.add_argument(
+    '--verbose',
+    action='store_true',
+    help='Whether or not to generate statistical report',
+)
+parser.add_argument('--device', type=str, default='cuda')
+parser.add_argument('-e', '--epochs', type=int, default=10,
+                    help='number of training epochs.')
+parser.add_argument('--num_layers', type=int, default=3,
+                    help='number of layers.')
+parser.add_argument('--num_heads', type=int, default=2,
+                    help='number of heads for GAT model.')
+parser.add_argument('-b', '--batch_size', type=int, default=1024,
+                    help='batch size.')
+parser.add_argument('--num_workers', type=int, default=12,
+                    help='number of workers.')
+parser.add_argument('--fan_out', type=int, default=10,
+                    help='number of neighbors in each layer')
+parser.add_argument('--hidden_channels', type=int, default=256,
+                    help='number of hidden channels.')
+parser.add_argument('--lr', type=float, default=0.003)
+parser.add_argument('--wd', type=float, default=0.0,
+                    help='weight decay for the optimizer')
+parser.add_argument('--dropout', type=float, default=0.5)
+parser.add_argument(
+    '--use_directed_graph',
+    action='store_true',
+    help='Whether or not to use directed graph',
+)
+args = parser.parse_args()
+if "papers" in str(args.dataset) and (psutil.virtual_memory().total /
+                                      (1024**3)) < 390:
+    print("Warning: may not have enough RAM to use this many GPUs.")
+    print("Consider upgrading RAM if an error occurs.")
+    print("Estimated RAM Needed: ~390GB.")
+verbose = args.verbose
+if verbose:
+    wall_clock_start = time.perf_counter()
+    if args.use_gat:
+        print(f"Training {args.dataset} with GAT model.")
+    else:
+        print(f"Training {args.dataset} with GraphSage model.")
+
+if not torch.cuda.is_available():
+    args.device = "cpu"
+device = torch.device(args.device)
+
+num_epochs = args.epochs
+num_layers = args.num_layers
+num_workers = args.num_workers
+num_hidden_channels = args.hidden_channels
+batch_size = args.batch_size
+
+root = osp.join(args.dataset_dir, args.dataset_subdir)
+print('The root is: ', root)
+dataset = PygNodePropPredDataset(name=args.dataset, root=root)
+split_idx = dataset.get_idx_split()
+evaluator = Evaluator(name=args.dataset)
```

Comment:
`evaluator` isn't used anywhere. Mind checking again to make sure there're no other unused variable?

Reply:
```suggestion
```